test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+                .url
+            }
+          )
+        }
+
+        "a user has answered the question before" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              sessionWithSingleDisposalState(
+                Some(IncompleteSupportingEvidenceAnswers(Some(false), List.empty, List.empty))
+              )._1
+            )
+          }
+
+          checkPageIsDisplayed(
+            performAction(),
+            messageFromMessageKey("supporting-evidence.do-you-want-to-upload.title"), { doc =>
+              doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.SupportingEvidenceController
+                .doYouWantToUploadSupportingDocuments()
+                .url
+            }
+          )
+        }
+      }
+    }
+
+    "handling submission of do you want to upload supporting evidence page" must {
+
+      def performAction(data: (String, String)*): Future[Result] =
+        controller.doYouWantToUploadSupportingDocumentsSubmit()(FakeRequest().withFormUrlEncodedBody(data: _*))
+
+      "display the technical error page" when {
+
+        "an error occurs when storing the draft return and answer is true" in {
+          val answers =
+            IncompleteSupportingEvidenceAnswers.empty.copy(
+              doYouWantToUploadSupportingEvidence = Some(true),
+              List.empty,
+              List.empty
+            )
+
+          val (session, journey, draftReturn) = sessionWithSingleDisposalState(Some(answers))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              draftReturn,
+              journey.subscribedDetails.cgtReference,
+              journey.agentReferenceNumber
+            )(Left(Error("some error")))
+          }
+
+          checkIsTechnicalErrorPage(performAction("supporting-evidence.do-you-want-to-upload" -> "true"))
+        }
+
+        "an error occurs when storing the draft return and answer is false" in {
+          val answers =
+            IncompleteSupportingEvidenceAnswers.empty.copy(
+              doYouWantToUploadSupportingEvidence = Some(false),
+              List.empty,
+              List.empty
+            )
+
+          val (session, journey, draftReturn) = sessionWithSingleDisposalState(Some(answers))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              draftReturn,
+              journey.subscribedDetails.cgtReference,
+              journey.agentReferenceNumber
+            )(Left(Error("some error")))
+          }
+
+          checkIsTechnicalErrorPage(performAction("supporting-evidence.do-you-want-to-upload" -> "false"))
+        }
```

Comment:
you could also test what happens if something goes wrong when updating the session